lib/profile.go

```diff
@@ -20,46 +18,86 @@ import (
 // changes. all of this code should be refactored into subroutines of general
 // configuration getters & setters
 type ProfileMethods struct {
-	inst *Instance
+	d dispatcher
 }
 
-// CoreRequestsName implements the Request interface
-func (ProfileMethods) CoreRequestsName() string { return "profile" }
+// Name returns the name of this method group
+func (m ProfileMethods) Name() string {
+	return "profile"
+}
 
-// NewProfileMethods creates a ProfileMethods pointer from either a repo
-// or an rpc.Client
-func NewProfileMethods(inst *Instance) *ProfileMethods {
-	return &ProfileMethods{inst: inst}
+// Attributes defines attributes for each method
+func (m ProfileMethods) Attributes() map[string]AttributeSet {
+	return map[string]AttributeSet{
+		"getprofile":      {denyRPC, ""},
+		"setprofile":      {denyRPC, ""},
+		"setprofilephoto": {denyRPC, ""},
+		"setposterphoto":  {denyRPC, ""},
+	}
 }
 
+type ProfileParams struct{}
```

Comment:
Please add a comment for this struct, to make `go lint` happy.

Reply:
ooof ty